### ISSUE TITLE
[BUG] Python analysis failing with invalid expression

### ISSUE THREAD

**Issue by sshveta** [konveyor/analyzer-lsp#1093]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

upstream

### Priority

Critical

### Current Behavior

Create an application and test with rule below : 

Rule 
=============
```
- ruleID: python-sample-rule-001
  message: python sample rule 001
  description: sample-rule-1 - this should appear as an issue
  category: potential
  effort: 1
  when:
    python.referenced:
      pattern: "hello_world"
- ruleID: python-sample-rule-002
  message: python sample rule 002
  description: sample-rule-2 - this should appear as an issue
  category: potential
  effort: 5
  when:
    python.referenced:
      pattern: "speak"
- ruleID: python-sample-rule-003
  message: python sample rule 003
  category: potential
  when:
    python.referenced:
      pattern: "create_custom_resource_definition"

```

Logs 
=============
```
id: 515
createUser: admin
createTime: 2026-02-07T00:34:03.21523552Z
name: pythonApp_Source_test-app-inventore51.420.analyzer
kind: analyzer
addon: analyzer
extensions:
    - python
state: Failed
priority: 10
data:
    mode:
        artifact: ""
        binary: false
        withDeps: false
    rules:
        labels:
            excluded: []
            included: []
        path: /rules
    scope:
        packages:
            excluded: []
            included: []
        withKnownLibs: false
    sources: []
    tagger:
        enabled: true
    targets: []
    verbosity: 1
application:
    id: 420
    name: pythonApp_Source_test-app-inventore51
bucket:
    id: 1123
    name: ""
pod: openshift-mta/task-515-cst26
started: 2026-02-07T00:34:03.454338945Z
terminated: 2026-02-07T00:34:09.120248545Z
events:
    - kind: AddonSelected
      count: 1
      last: 2026-02-07T00:34:03.402635046Z
    - kind: ExtensionSelected
      count: 1
      reason: python
      last: 2026-02-07T00:34:03.449477752Z
    - kind: PodCreated
      count: 1
      reason: openshift-mta/task-515-cst26
      last: 2026-02-07T00:34:03.545619939Z
    - kind: PodPending
      count: 2
      reason: ContainerCreating
      last: 2026-02-07T00:34:04.606094847Z
    - kind: PodRunning
      count: 4
      last: 2026-02-07T00:34:09.120247764Z
    - kind: PodFailed
      count: 1
      reason: Error
      last: 2026-02-07T00:34:09.120249557Z
    - kind: ContainerKilled
      count: 1
      reason: 'container: ''python'' had not terminated.'
      last: 2026-02-07T00:34:09.285891471Z
errors:
    - severity: Error
      description: 'Container (addon) failed: Error'
    - severity: Error
      description: invalid expression ''
activity:
    - 'OptDir:    /addon/opt'
    - 'SharedDir: /shared'
    - 'CacheDir:  /cache'
    - 'SourceDir: /shared/source'
    - 'RuleDir:   /addon/rules'
    - 'BinDir:    /shared/bin'
    - 'M2Dir:     /cache/m2'
    - Fetching application.
    - '[CMD] Running: /usr/bin/rm -rf /addon/.git/8e14110b9247e893'
    - '[CMD] /usr/bin/rm succeeded.'
    - '[GIT] Created: /addon/.git/8e14110b9247e893/.gitconfig'
    - '[CMD] Running: /usr/bin/git clone https://github.com/konveyor/analyzer-lsp /shared/source/analyzer-lsp'
    - '[CMD] /usr/bin/git succeeded.'
    - '[RULESET] fetching bucket: /rules'
attached:
    - id: 3403
      name: rm.output
      activity: 9
    - id: 3404
      name: git.output
      activity: 12
    - id: 3405
      name: addon.log
    - id: 3406
      name: python.log
    - id: 3408
      name: pod.yaml

```

<img width="1996" height="668" alt="Image" src="https://github.com/user-attachments/assets/99bb1d57-af14-4816-bfc4-ec1e020082f2" />

### How Reproducible

Always (Default)

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by jmle** [comment]
I have been taking a quick look and am not sure if this is the analyzer's fault. It's coming from an empty label being sent in the custom rule struct, but the rule yaml itself doesn't have any labels. In the analyzer's python e2e tests we have [similar rules](https://github.com/konveyor/analyzer-lsp/blob/main/external-providers/generic-external-provider/e2e-tests/python-e2e/rule-example.yaml), which also don't use labels, but they are passing.

Could this be the UI passing an explicit empty label?
